Add media attachment filter and integration tests

pfefferle opened the pull request:
Introduces the filter_attached_media_ids method to Classic_Editor for limiting and retrieving image attachments from a post's media library, and hooks it into the activitypub_attachment_ids filter. Adds comprehensive integration tests for Classic_Editor, including tests for media markup, attachment limits, meta data saving, and permission checks. Test assets for audio, image, and video are also added for use in PHPUnit tests.

## Proposed changes:
- Added comprehensive test coverage for Classic Editor integration
- Added new filter hook `activitypub_attachment_ids` to retrieve attached media from post library
- Added test asset files (sample video, audio, and image) to support attachment testing

### Other information:

- [x] Have you written new tests for your changes, if applicable?

## Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

* Go to '..'
*

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box below and supplying data. -->
<!-- It will trigger a GitHub workflow that will create and push the entry into the branch. -->

<!-- Due to org permissions, the job may fail for PRs crated from a fork under GitHub organizations. -->
<!-- In this case, you can create entry manually with `composer changelog:add` and push it into the branch. -->

<!-- If no changelog entry is required for this PR, please add the "Skip Changelog" label. -->

-   [X] Automatically create a changelog entry from the details below.

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [ ] Patch
-   [X] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Added - for new features
-   [X] Changed - for changes in existing functionality
-   [ ] Deprecated - for soon-to-be removed features
-   [ ] Removed - for now removed features
-   [ ] Fixed - for any bug fixes
-   [ ] Security - in case of vulnerabilities

#### Message

Improved Classic Editor integration by adding better media handling and full test coverage for attachments, permissions, and metadata.

</details>

pfefferle committed:
Add media attachment filter and integration tests

Introduces the filter_attached_media_ids method to Classic_Editor for limiting and retrieving image attachments from a post's media library, and hooks it into the activitypub_attachment_ids filter. Adds comprehensive integration tests for Classic_Editor, including tests for media markup, attachment limits, meta data saving, and permission checks. Test assets for audio, image, and video are also added for use in PHPUnit tests.

pfefferle committed:
Merge branch 'trunk' into add-attachment-support

matticbot committed:
Add changelog

obenland commented on tests/phpunit/tests/integration/class-test-classic-editor.php:
```diff
@@ -0,0 +1,479 @@
+
+		parent::tear_down_after_class();
+	}
+
```
```suggestion
```
This should be covered by the parent tear down method

obenland commented on tests/phpunit/tests/integration/class-test-classic-editor.php:
```diff
@@ -0,0 +1,479 @@
+	 */
+	public function test_filter_attachments_media_markup_single_video() {
+		$video_id = self::factory()->attachment->create_upload_object(
+			ACTIVITYPUB_PLUGIN_DIR . 'tests/phpunit/data/assets/sample-video.mp4',
```
```suggestion
			AP_TESTS_DIR . '/data/assets/sample-video.mp4',
```

obenland commented on tests/phpunit/tests/integration/class-test-classic-editor.php:
```diff
@@ -0,0 +1,479 @@
+	 */
+	public function test_filter_attachments_media_markup_single_audio() {
+		$audio_id = self::factory()->attachment->create_upload_object(
+			ACTIVITYPUB_PLUGIN_DIR . 'tests/phpunit/data/assets/sample-audio.mp3',
```
```suggestion
			AP_TESTS_DIR . '/data/assets/sample-audio.mp3',
```

obenland commented on tests/phpunit/tests/integration/class-test-classic-editor.php:
```diff
@@ -0,0 +1,479 @@
+	 */
+	public function test_filter_attachments_media_markup_multiple_images() {
+		$image_id_1 = self::factory()->attachment->create_upload_object(
+			ACTIVITYPUB_PLUGIN_DIR . 'tests/phpunit/data/assets/sample-image.jpg',
```
```suggestion
			AP_TESTS_DIR . '/data/assets/sample-image.jpg',
```

obenland commented on tests/phpunit/tests/integration/class-test-classic-editor.php:
```diff
@@ -0,0 +1,479 @@
+			self::$post_id
+		);
+		$image_id_2 = self::factory()->attachment->create_upload_object(
+			ACTIVITYPUB_PLUGIN_DIR . 'tests/phpunit/data/assets/sample-image.jpg',
```
```suggestion
			AP_TESTS_DIR . '/data/assets/sample-image.jpg',
```

obenland commented on tests/phpunit/tests/integration/class-test-classic-editor.php:
```diff
@@ -0,0 +1,479 @@
+	 */
+	public function test_filter_attachments_media_markup_single_image() {
+		$image_id = self::factory()->attachment->create_upload_object(
+			ACTIVITYPUB_PLUGIN_DIR . 'tests/phpunit/data/assets/sample-image.jpg',
```
```suggestion
			AP_TESTS_DIR . '/data/assets/sample-image.jpg',
```

obenland commented on tests/phpunit/tests/integration/class-test-classic-editor.php:
```diff
@@ -0,0 +1,479 @@
+	public function test_filter_attached_images_with_attachments() {
+		// Create image attachments.
+		$image_id_1 = self::factory()->attachment->create_upload_object(
+			ACTIVITYPUB_PLUGIN_DIR . 'tests/phpunit/data/assets/sample-image.jpg',
```
```suggestion
			AP_TESTS_DIR . '/data/assets/sample-image.jpg',
```

obenland commented on tests/phpunit/tests/integration/class-test-classic-editor.php:
```diff
@@ -0,0 +1,479 @@
+			self::$post_id
+		);
+		$image_id_2 = self::factory()->attachment->create_upload_object(
+			ACTIVITYPUB_PLUGIN_DIR . 'tests/phpunit/data/assets/sample-image.jpg',
```
```suggestion
			AP_TESTS_DIR . '/data/assets/sample-image.jpg',
```

obenland commented on tests/phpunit/tests/integration/class-test-classic-editor.php:
```diff
@@ -0,0 +1,479 @@
+		$image_ids = array();
+		for ( $i = 0; $i < 5; $i++ ) {
+			$image_ids[] = self::factory()->attachment->create_upload_object(
+				ACTIVITYPUB_PLUGIN_DIR . 'tests/phpunit/data/assets/sample-image.jpg',
```
```suggestion
				AP_TESTS_DIR . '/data/assets/sample-image.jpg',
```

obenland commented on tests/phpunit/tests/integration/class-test-classic-editor.php:
```diff
@@ -0,0 +1,479 @@
+		$image_ids = array();
+		for ( $i = 0; $i < 3; $i++ ) {
+			$image_ids[] = self::factory()->attachment->create_upload_object(
+				ACTIVITYPUB_PLUGIN_DIR . 'tests/phpunit/data/assets/sample-image.jpg',
```
```suggestion
				AP_TESTS_DIR . '/data/assets/sample-image.jpg',
```

obenland commented on tests/phpunit/tests/integration/class-test-classic-editor.php:
```diff
@@ -0,0 +1,479 @@
+		$image_ids = array();
+		for ( $i = 0; $i < 3; $i++ ) {
+			$image_ids[] = self::factory()->attachment->create_upload_object(
+				ACTIVITYPUB_PLUGIN_DIR . 'tests/phpunit/data/assets/sample-image.jpg',
```
```suggestion
				AP_TESTS_DIR . '/data/assets/sample-image.jpg',
```

pfefferle committed:
Update tests/phpunit/tests/integration/class-test-classic-editor.php

Co-authored-by: Konstantin Obenland <obenland@gmx.de>

pfefferle committed:
Update tests/phpunit/tests/integration/class-test-classic-editor.php

Co-authored-by: Konstantin Obenland <obenland@gmx.de>

pfefferle committed:
Update tests/phpunit/tests/integration/class-test-classic-editor.php

Co-authored-by: Konstantin Obenland <obenland@gmx.de>

pfefferle committed:
Update tests/phpunit/tests/integration/class-test-classic-editor.php

Co-authored-by: Konstantin Obenland <obenland@gmx.de>

pfefferle committed:
Update tests/phpunit/tests/integration/class-test-classic-editor.php

Co-authored-by: Konstantin Obenland <obenland@gmx.de>

pfefferle committed:
Update tests/phpunit/tests/integration/class-test-classic-editor.php

Co-authored-by: Konstantin Obenland <obenland@gmx.de>

pfefferle committed:
Update tests/phpunit/tests/integration/class-test-classic-editor.php

Co-authored-by: Konstantin Obenland <obenland@gmx.de>

pfefferle committed:
Update test image path to use AP_TESTS_DIR constant

Replaces usage of ACTIVITYPUB_PLUGIN_DIR with AP_TESTS_DIR for test image paths in class-test-classic-editor.php. This improves consistency and may better reflect the intended test directory structure.

pfefferle committed:
Merge branch 'trunk' into add-attachment-support

pfefferle committed:
Fix image path in classic editor integration tests

Updated the path to sample-image.jpg in multiple test cases to correct the directory structure. This ensures that the tests reference the correct file location for attachments.

obenland commented on integration/class-classic-editor.php:
```diff
@@ -51,6 +52,51 @@ public static function filter_attachments_media_markup( $markup, $attachment_ids
+			return array();
+		}
+
+		$query = new \WP_Query(
```
Have you considered `get_attached_media( 'image', $item->ID )`?

obenland replied:
Hm, I guess it doesn't allow limiting the number.

pfefferle committed:
Merge branch 'trunk' into add-attachment-support

pfefferle committed:
Fix media attachment merging in Classic Editor

Corrects the logic in filter_attached_media_ids to merge new media IDs with existing attachments and return the full list up to the configured maximum. Updates related tests to reflect the new behavior, ensuring existing attachments are preserved and the total does not exceed the maximum.

pfefferle commented:
@obenland had to add some small tweaks... ready for another review...
